src/views/cards/CompleteAddCard/index.tsx

```diff
@@ -1,30 +1,84 @@
-import React from "react";
+import React, { useEffect, useMemo } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
 
-import { Card } from "@/components/cards";
-import type { CardInfo } from "@/components/cards/Card/Card";
-import { Button, Input } from "@/components/common";
+import { Card, CardNameInput } from "@/components/cards";
+import { useCardNameInput } from "@/components/cards/CardNameInput/hook";
+import { Button } from "@/components/common";
+import { CardItem, useCardsContext } from "@/contexts/CardsContext";
 import { CardPageLayout } from "@/layouts/cards";
+import { domains } from "@/router";
 
 import * as S from "./completeAddCard";
 
-const mockCardInfo: CardInfo = {
-  cardName: "",
-  cardOwnerName: "LeeJaeyun",
-  expireDate: "11/22",
-  cardNumber: "1111-1111-1111-1111",
-};
-
 export default function CompleteAddCard() {
+  const { state, pathname } = useLocation();
+  const navigate = useNavigate();
+
+  const { dispatch } = useCardsContext();
+
+  const { cardName, onCardNameChange } = useCardNameInput({
+    cardName: state?.cardInfo?.cardName ?? "",
+  });
+
+  useEffect(() => {
+    if (!state?.cardInfo) navigate(domains.CARD_LIST);
+  }, []);
```

Comment:
```suggestion
  useRedirectCardList(state?.cardInfo == null);
```
이렇게 의도한 상태가 아닐 때 홈으로 돌아가는 custom hook을 만들어두면 좋을 것 같아요~